Rules.py

```diff
@@ -371,38 +451,184 @@ def global_rules(world, player):
 
+def bomb_rules(world, player):
+    bonkable_doors = ['Two Brothers House Exit (West)', 'Two Brothers House Exit (East)'] # Technically this is incorrectly defined, but functionally the same as what is intended.
+    bombable_doors = ['Ice Rod Cave', 'Light World Bomb Hut', 'Light World Death Mountain Shop', 'Mini Moldorm Cave',
+                      'Hookshot Cave Exit (South)', 'Hookshot Cave Exit (North)', 'Dark Lake Hylia Ledge Fairy', 'Hype Cave', 'Brewery']
+    for entrance in bonkable_doors:
+        add_rule(world.get_entrance(entrance, player), lambda state: state.can_use_bombs(player) or state.has_Boots(player)) 
+    for entrance in bombable_doors:
+        add_rule(world.get_entrance(entrance, player), lambda state: state.can_use_bombs(player)) 
+
+    bonkable_items = ['Sahasrahla\'s Hut - Left', 'Sahasrahla\'s Hut - Middle', 'Sahasrahla\'s Hut - Right']
+    bombable_items = ['Blind\'s Hideout - Top', 'Kakariko Well - Top', 'Chicken House', 'Aginah\'s Cave', 'Graveyard Cave',
+                      'Paradox Cave Upper - Left', 'Paradox Cave Upper - Right',
+                      'Hype Cave - Top', 'Hype Cave - Middle Right', 'Hype Cave - Middle Left', 'Hype Cave - Bottom']
+    for location in bonkable_items:
+        add_rule(world.get_location(location, player), lambda state: state.can_use_bombs(player) or state.has_Boots(player)) 
+    for location in bombable_items:
+        add_rule(world.get_location(location, player), lambda state: state.can_use_bombs(player)) 
+
+    cave_kill_locations = ['Mini Moldorm Cave - Far Left', 'Mini Moldorm Cave - Far Right', 'Mini Moldorm Cave - Left', 'Mini Moldorm Cave - Right', 'Mini Moldorm Cave - Generous Guy']
+    for location in cave_kill_locations:
+        add_rule(world.get_location(location, player), lambda state: state.can_kill_most_things(player) or state.can_use_bombs(player))
+
+    paradox_switch_chests = ['Paradox Cave Lower - Far Left', 'Paradox Cave Lower - Left', 'Paradox Cave Lower - Right', 'Paradox Cave Lower - Far Right', 'Paradox Cave Lower - Middle']
+    for location in paradox_switch_chests:
+        add_rule(world.get_location(location, player), lambda state: state.can_hit_crystal_through_barrier(player))
+
+    # Dungeon bomb logic
+    easy_kill_rooms = [ # Door, bool-bombable
+        ('Hyrule Dungeon Armory S', True), # One green guard
+        ('Hyrule Dungeon Armory ES', True), # One green guard
+        ('Hyrule Dungeon Armory Boomerang WS', True), # One blue guard
+        ('Desert Compass NW', True), # Three popos
+        ('Desert Four Statues NW', True), # Four popos
+        ('Desert Four Statues ES', True), # Four popos
+        ('Hera Beetles WS', False), # Three blue beetles and only two pots, and bombs don't work.
+        ('Thieves Basement Block WN', True), # One blue and one red zazak and one Stalfos. Two pots. Need to kill the third enemy somehow.
+        ('Ice Pengator Trap NE', False), # Five pengators. Bomb-doable?
```

Comment:
Yeah, Pengators die to one bomb and you can kill the whole line-up with only two bombs with good placement. Bomb strat is even faster if you are lacking hookshot or decent sword.

Reply:
These rules do not work with enemizer (probably should default to can_kill_most_things, maybe in that case)